config/always-python.m4

```diff
@@ -46,42 +57,36 @@ AC_DEFUN([ZFS_AC_CONFIG_ALWAYS_PYTHON], [
 		[with_python=check])
 
 	AS_CASE([$with_python],
-		[check],
-		[AS_IF([test -x /usr/bin/python3],
-			[PYTHON="python3"],
-			[AS_IF([test -x /usr/bin/python2],
-				[PYTHON="python2"],
-				[PYTHON=""]
-			)]
-		)],
+		[check], [AC_CHECK_PROGS([PYTHON], [python3 python2], [:])],
 		[2*], [PYTHON="python${with_python}"],
 		[*python2*], [PYTHON="${with_python}"],
 		[3*], [PYTHON="python${with_python}"],
 		[*python3*], [PYTHON="${with_python}"],
-		[no], [PYTHON=""],
+		[no], [PYTHON=":"],
 		[AC_MSG_ERROR([Unknown --with-python value '$with_python'])]
 	)
 
-	AS_IF([$PYTHON --version >/dev/null 2>&1], [ /bin/true ], [
-		AC_MSG_ERROR([Cannot find $PYTHON in your system path])
+	AS_IF([test $PYTHON != :], [
+		AS_IF([$PYTHON --version >/dev/null 2>&1],
+			[AM_PATH_PYTHON([2.6], [], [:])],
```

Comment:
This check is actually redundant, since AM_PATH_PYTHON will already try to run $PYTHON -c some-inline-python-script-which-consults-sys.hexversion.

Reply:
Okay, a bit late, hehe.

Reply:
Indeed, well to remove the redundancy let's open a new PR.